### PR TITLE
Feat/delegate subnet takes cli

### PR DESCRIPTION
**Pull request by distributedstatemachine**
## Description

This PR:

- add  `btcli root set_delegate_takes  --netuids "1,2" --takes "0.1,0.15"` so that delegatees can set the stakes across multiple subnets at once

### CODE REVIEW

File: bittensor/commands/delegates.py [hunk 1]
```diff
@@ -1031,3 +1030,157 @@ def check_config(config: "bittensor.config"):
...
+                )
+                + " ..."
+            )
+            cli.config.takes = Prompt.ask(f"Enter takes (e.g. {example})")
+
+        # Parse from string
+        netuids_input = torch.tensor(
+            list(map(int, re.split(r"[ ,]+", cli.config.netuids))), dtype=torch.long
+        )
+        takes_input = torch.tensor(
```
Comment (thealligatorking) [inline bittensor/commands/delegates.py]:
```
takes_input = np.array(
    list(map(float, re.split(r"[ ,]+", cli.config.takes))),
    dtype=np.float32,
)
```
Faster if you use numpy

**Review by thealligatorking — APPROVED** [review]

File: bittensor/commands/delegates.py [hunk 2]
```diff
@@ -1031,3 +1030,157 @@ def check_config(config: "bittensor.config"):
...
+            SetDelegateTakesCommand._run(cli, subtensor)
+        finally:
+            if "subtensor" in locals():
+                subtensor.close()
+                bittensor.logging.debug("closing subtensor connection")
+
+    @staticmethod
+    def _run(cli: "bittensor.cli", subtensor: "bittensor.subtensor"):
+        r"""Set takes for multiple subnets."""
+        # config = cli.config.copy()
```
Comment (thealligatorking) [inline bittensor/commands/delegates.py]: NIT: debug

File: bittensor/extrinsics/delegation.py [hunk 3]
```diff
@@ -497,3 +499,70 @@ def increase_take_extrinsic(
...
+        try:
+            success = subtensor._set_delegate_takes(
+                wallet=wallet,
+                hotkey_ss58=hotkey_ss58,
+                takes=takes,
+                wait_for_inclusion=wait_for_inclusion,
+                wait_for_finalization=wait_for_finalization,
+            )
+
+            if success == True:
```
Comment (thealligatorking) [inline bittensor/extrinsics/delegation.py]: Truthy can just be `if success:`

File: bittensor/subtensor.py [hunk 4]
```diff
@@ -4837,6 +4919,99 @@ def make_substrate_call_with_retry():
 
         return make_substrate_call_with_retry()
 
+    # def _set_delegate_takes(
```
Comment (thealligatorking) [inline bittensor/subtensor.py]: we should rm before merge if not needed

File: bittensor/extrinsics/delegation.py (hunk 3 above)
Comment (distributedstatemachine) [reply bittensor/extrinsics/delegation.py]: resolved

**Pull request merged**